src/main/java/org/apache/commons/imaging/ImageParser.java

```diff
@@ -954,13 +938,12 @@ protected final boolean canAcceptExtension(final String filename) {
      * @param params A valid Map object, or a null.
      * @return A valid instance of an implementation of a IBufferedImageFactory.
      */
-    protected BufferedImageFactory getBufferedImageFactory(final Map<String, Object> params) {
-        if (params == null) {
+    protected BufferedImageFactory getBufferedImageFactory(final ImagingParameters params) {
+        if (params == null || params.isBufferedImageFactoryPresent() == false) {
             return new SimpleBufferedImageFactory();
         }
 
-        final BufferedImageFactory result = (BufferedImageFactory) params
-                .get(ImagingConstants.BUFFERED_IMAGE_FACTORY);
+        final BufferedImageFactory result = params.getBufferedImageFactory();
```

Comment:
As noted in JIRA, to me it feels strange to pass a BufferedImageFactory via a parameters object. But since this has been the case before this PR, I think we should first concentrate on getting the ImagingParameters in place and refactor the BufferedImageFactory later (if that is possible)